### DISCUSSION TITLE
What is the meaning of this comment mark ($NON-NLS-1$) in the source code?

### DISCUSSION THREAD

**Discussion by bianyun1981** [dbeaver/dbeaver#15277] [DBeaver Development]
There are a lot of comment tags like $NON-NLS-1$ in the source code. 

What do these tags mean? Are the hardcoded strings that appear in the marked source lines not needed or can't be localized, or are they not localized for the time being and are just marked to make them easier to find later?

```
org.jkiss.dbeaver.ui.navigator.dialogs.SelectDataSourceDialog
  - line 317-318:
        createButton(composite, IDialogConstants.OK_ID, "&Select", true); //$NON-NLS-1$
        createButton(composite, IDialogConstants.IGNORE_ID, "&None", false); //$NON-NLS-1$
```

**Comment by LonwoLonwo** [comment]
Hello @bianyun1981 

> lines not needed or can't be localized

This is the right answer.

**Reply by bianyun1981** [reply]
Thanks for your reply, now I know which of the hardcoded strings are not to care about
